Modules/fcntlmodule.c

```diff
@@ -228,14 +204,14 @@ fcntl_ioctl_impl(PyObject *module, int fd, unsigned int code,
         if (PyArg_Parse(ob_arg, "s*:ioctl", &pstr)) {
             str = pstr.buf;
             len = pstr.len;
-            if (len > IOCTL_BUFSZ) {
-                PyBuffer_Release(&pstr);
-                PyErr_SetString(PyExc_ValueError,
-                                "ioctl string arg too long");
+
+            PyObject *o = PyBytes_FromStringAndSize(NULL, len);
```

Comment:
it's not decref in the failure/error code path(s)..

also on line 248 : could not you directly returns it instead of re-creating another object ?